Sources/AWSSigner/signer.swift

```diff
@@ -12,18 +12,18 @@ import NIO
 import NIOHTTP1
 
 /// Amazon Web Services V4 Signer
-public final class AWSSigner {
+public struct AWSSigner {
     /// security credentials for accessing AWS services
-    public let credentials: CredentialProvider
+    public var credentials: Credential
     /// service signing name. In general this is the same as the service name
-    public let name: String
+    public var name: String
     /// AWS region you are working in
-    public let region: String
+    public var region: String
```

Comment:
I wouldn't make these properties mutable. What would the usecase be?

Reply:
reverted, again was trying something forgot to revert